### PR TITLE
Convert logging to standard JSON format.

### PR DESCRIPTION
**Pull request by DanielChicot**

### CODE REVIEW

File: src/main/kotlin/LoggerUtils.kt [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        builder.append(event.level)
+        builder.append("\", \"message\":\"")
+        builder.append(flattenMultipleLines(event.formattedMessage))
+        builder.append("\", ")
+        builder.append(throwableProxyEventToString(event))
+        builder.append("\"thread\":\"")
+        builder.append(event.threadName)
+        builder.append("\", \"logger\":\"")
+        builder.append(event.loggerName)
+        builder.append("\", \"duration_in_milliseconds\":\"")
```
Comment (steveburton4) [inline src/main/kotlin/LoggerUtils.kt]: Can remove duration_in_milliseconds and the method below, it's not a thing in K2HB.

File: src/main/kotlin/LoggerUtils.kt [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+private const val UNSET_TEXT = "NOT_SET"
+private val defaultFormat = makeUtcDateFormat() // 2001-07-04T12:08:56.235
+
+private var hostname = InetAddress.getLocalHost().hostName
+private var environment = System.getProperty("environment", UNSET_TEXT)
+private var application = System.getProperty("application", UNSET_TEXT)
+private var app_version = System.getProperty("app_version", UNSET_TEXT)
+private var component = System.getProperty("component", UNSET_TEXT)
+var correlation_id: String = System.getProperty("correlation_id", UNSET_TEXT)
```
Comment (steveburton4) [inline src/main/kotlin/LoggerUtils.kt]: I don't think correlation_id is needed in K2HB, I'd remove it.

File: src/main/kotlin/LoggerUtils.kt [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+import org.slf4j.LoggerFactory
+import java.net.InetAddress
+import java.text.SimpleDateFormat
+import java.util.*
+
+private const val UNSET_TEXT = "NOT_SET"
+private val defaultFormat = makeUtcDateFormat() // 2001-07-04T12:08:56.235
+
+private var hostname = InetAddress.getLocalHost().hostName
+private var environment = System.getProperty("environment", UNSET_TEXT)
```
Comment (steveburton4) [inline src/main/kotlin/LoggerUtils.kt]: We don't set any system properties in K2HB I think, we send in as env vars.

File: src/main/kotlin/HbaseClient.kt [hunk 4]
```diff
@@ -59,19 +58,19 @@ open class HbaseClient(val connection: Connection, val columnFamily: ByteArray,
...
+                logger.info("Namespace already exists, probably created by another process", "namespace", namespace)
             }
             finally {
                 namespaces[namespace] = true
             }
         }
 
         if (!tables.contains(tableName)) {
-            logger.info("Creating table '$dataTableName'.")
+            logger.info("Creating table", "dataTableName", "$dataTableName")
```
Comment (mark-r-m) [inline src/main/kotlin/HbaseClient.kt]: _name fits how we are doing the key names in the other apps?

File: src/main/kotlin/HbaseClient.kt [hunk 5]
```diff
@@ -82,7 +81,8 @@ open class HbaseClient(val connection: Connection, val columnFamily: ByteArray,
                             })
                 })
             } catch (e: TableExistsException) {
-                logger.info("'$tableName' already exists, probably created by another process")
+                logger.info("Didn't create table, table already exists, probably created by another process",
+                    "tableName", tableName)
```
Comment (mark-r-m) [inline src/main/kotlin/HbaseClient.kt]: table_name ??

File: src/main/kotlin/LoggerUtils.kt (hunk 1 above)
Comment (mark-r-m) [reply src/main/kotlin/LoggerUtils.kt]: Or keep it - I'd like to know what we think the poll time etc is

File: src/main/kotlin/Config.kt [hunk 6]
```diff
@@ -84,10 +84,7 @@ object Config {
         var topicRegex: Pattern = Pattern.compile(getEnv("K2HB_KAFKA_TOPIC_REGEX") ?: "db.*")
         var dlqTopic = getEnv("K2HB_KAFKA_DLQ_TOPIC") ?: "dead-letter-queue"
 
-        fun reportTopicSubscriptionDetails(): String {
```
Comment (mark-r-m) [inline src/main/kotlin/Config.kt]: I think we should keep some of this?

**Comment by mark-r-m** [comment]
Also - can we add logging of our offsets periodically???

File: src/main/kotlin/HbaseClient.kt (hunk 4 above)
Comment (DanielChicot) [reply src/main/kotlin/HbaseClient.kt]: changed

File: src/main/kotlin/HbaseClient.kt (hunk 5 above)
Comment (DanielChicot) [reply src/main/kotlin/HbaseClient.kt]: changed

File: src/main/kotlin/Config.kt (hunk 6 above)
Comment (DanielChicot) [reply src/main/kotlin/Config.kt]: Everything that it returned is logged in subsequent messages. Previously it was all logged twice.

File: src/main/kotlin/LoggerUtils.kt (hunk 1 above)
Comment (steveburton4) [reply src/main/kotlin/LoggerUtils.kt]: Yeah agreed if it is set and looks good

File: src/main/kotlin/LoggerUtils.kt (hunk 3 above)
Comment (DanielChicot) [reply src/main/kotlin/LoggerUtils.kt]: Used env vars. Incoming change in aws-ingest to set these env vars.

File: src/main/kotlin/LoggerUtils.kt (hunk 1 above)
Comment (DanielChicot) [reply src/main/kotlin/LoggerUtils.kt]: I'll leave it in as at least 1 person thinks it will be useful.

**Pull request merged**